ISISNeutronMuon/MDANSE
# Updated the atom selector and added the front end atom selection helper

**Description of work**
Changed the atom selection mechanism so that it is now a selector and added a setting to invert the selection so a filter can still be done. Added a atom selection front end so that the setting can be created using the GUI.

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/34aabb54-272f-43db-8b30-5e64c2fafeef)


**To test**
Unit tests should pass.

In the GUI load a trajectory and open a analysis window with atom selection such as the position autocorrelation function and check that:
- the atom selection helper button opens the helper window
- clicking the atom selection helper button should close the window
- helper window should open next to the analysis window
- options in the atom selection helper should be greyed out if those group/molecules/elements are not seen in the chosen trajectory
- select all in the combobox should behave in the expected way e.g. unchecking select all deselects all options.
- the textbox should update correctly after selecting different selections
- clicking apply should update the atom_selection field in the analysis window
- editing the atom_selection value in the analysis window should change black/red on a correct/incorrect setting

After applying the atom selection setting run an analysis and check that the atom selection works e.g. remove a specific oxygen and there are no oxygen atoms in you results.
